### ISSUE TITLE
VSCode not suggesting the component's props

### ISSUE THREAD

**Issue by thanhloc-17** [wix/react-native-ui-lib#2230]
This problem occur when I update new version or RNUILib, then I type some prop and VSCode not suggest any props
<img width="568" alt="image" src="https://user-images.githubusercontent.com/20339875/187686212-6067f88a-f3f4-492f-ae6f-2c0be1f8b2eb.png">

With the others package, the IntelliSense still working well.

Thanks

**Comment by ramssutharr** [comment]
having same issue on react native v0.70 .

**Comment by qkudev** [comment]
Me too!

**Comment by thanhloc-17** [comment]
up

**Comment by tarikpnr** [comment]
any news on this? @ethanshar

**Comment by lakardion** [comment]
Same on my side. I get key and ref as the only two available props
It is happening on ALL imported components from the library:
![image](https://user-images.githubusercontent.com/52294448/197536560-c06982da-8178-44dd-b4b4-80d60f509740.png)

**Comment by The-White-Fang** [comment]
any updates or hints on what might be wrong?

**Comment by The-White-Fang** [comment]
possible duplicate of https://github.com/wix/react-native-ui-lib/issues/2202 ?

**Comment by KishoreBarik** [comment]
same issue here

**Comment by thanhloc-17** [comment]
up

**Comment by thanhloc-17** [comment]
up

**Comment by chrismerino** [comment]
Issue still persists on RN 0.70.5 and react-native-ui-lib: 7.1.0

**Comment by ethanshar** [comment]
Hi, is this issue still relevant? 
I don't experience it

**Comment by jkaveri** [comment]
I still experience it on `7.3.5`

**Comment by jkaveri** [comment]
<img width="1205" alt="Screenshot 2023-05-14 at 11 45 32" src="https://github.com/wix/react-native-ui-lib/assets/884709/25a28694-bdc8-4e18-be17-087d9ced6a88">
It looks like the components is losing props type.

**Comment by The-White-Fang** [comment]
> Hi, is this issue still relevant? I don't experience it

Could you share your `tsconfig.json` please? We may have to add something to it to make it work. 
Starting a new project from scratch and installing the lib doesn't work for us. 

A workaround I found was to remove the global type `ThemeComponent` and place it in an `export` statement. But that seems like something that will never be in the official repo.
I am still looking into it and will let you know if I can find something else to fix it.

**Comment by jkaveri** [comment]
my tsconfig below:

```
{
  "extends": "expo/tsconfig.base",
  "compilerOptions": {
    "strict": true,
    "baseUrl": ".",
    "typeRoots": ["./node_modules/@types", "./@types"]
  }
}

```

I defined global ThemeComponent but it doesn't work.

**Comment by The-White-Fang** [comment]
> > Hi, is this issue still relevant? I don't experience it
> 
> Could you share your `tsconfig.json` please? We may have to add something to it to make it work. Starting a new project from scratch and installing the lib doesn't work for us.
> 
> A workaround I found was to remove the global type `ThemeComponent` and place it in an `export` statement. But that seems like something that will never be in the official repo. I am still looking into it and will let you know if I can find something else to fix it.
> 
> Edit: I noticed that generated `d.ts` files don't have any reference to `ThemeComponent`. ![image](https://user-images.githubusercontent.com/49617208/239745802-1a5fc1ab-920f-4787-b9af-866017e38a42.png)
> 
> Maybe that is the problem here.

I was right, it turns out that if you define your types in `d.ts` files, they will be excluded from your build. The solution to this would be to move the global types to imported types.

**Comment by M-i-k-e-l** [comment]
Should be fixed with the above PR

**Comment by Revolt9k** [comment]
Any news? Still experience it on `7.6.0`

**Comment by vicdelfant** [comment]
a885dfdb675025fd03d47119834f649462e056a4 was from Jul 24, 2023 and `7.6.0` is from before (July 20th), so we're still waiting for a new release.

**Comment by ethanshar** [comment]
Should be fixed in version `7.7.0` that should be released next week
